RenderManager_xbox/DistortionCorrectRenderManager.cpp

```diff
@@ -0,0 +1,760 @@
+    // code.
+    GLuint frameBuffer;               //< Groups a color buffer and a depth buffer
+    glGenFramebuffers(1, &frameBuffer);
+    glBindFramebuffer(GL_FRAMEBUFFER, frameBuffer);
+
+    for (size_t i = 0; i < renderInfo.size(); i++) {
+
+      // The color buffer for this eye.  We need to put this into
+      // a generic structure for the Present function, but we only need
+      // to fill in the OpenGL portion.
+      //  Note that this must be used to generate a RenderBuffer, not just
+      // a texture, if we want to be able to present it to be rendered
+      // via Direct3D for DirectMode.  This is selected based on the
+      // config file value, so we want to be sure to use the more general
+      // case.
+      //  Note that this texture format must be RGBA and unsigned byte,
+      // so that we can present it to Direct3D for DirectMode
+      GLuint colorBufferName = 0;
+      glGenRenderbuffers(1, &colorBufferName);
+      osvr::renderkit::RenderBuffer rb;
+      rb.OpenGL = new osvr::renderkit::RenderBufferOpenGL;
+      rb.OpenGL->colorBufferName = colorBufferName;
+      colorBuffers.push_back(rb);
+
+      // "Bind" the newly created texture : all future texture
+      // functions will modify this texture glActiveTexture(GL_TEXTURE0);
+      glBindTexture(GL_TEXTURE_2D, colorBufferName);
+
+      // Determine the appropriate size for the frame buffer to be used for
+      // this eye.
+      int width = static_cast<int>(renderInfo[i].viewport.width);
+      int height = static_cast<int>(renderInfo[i].viewport.height);
+
+      // Give an empty image to OpenGL ( the last "0" means "empty" )
+      glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA,
+        width,
+        height,
+        0,
+        GL_RGBA, GL_UNSIGNED_BYTE, 0);
+
+      // Bilinear filtering
+      glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+      glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+      glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+      glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+
+      // The depth buffer
+      GLuint depthrenderbuffer;
+      glGenRenderbuffers(1, &depthrenderbuffer);
+      glBindRenderbuffer(GL_RENDERBUFFER, depthrenderbuffer);
+      glRenderbufferStorage(GL_RENDERBUFFER, GL_DEPTH_COMPONENT,
+        width,
+        height);
+      depthBuffers.push_back(depthrenderbuffer);
+    }
+
+    // Register our constructed buffers so that we can use them for
+    // presentation.
+    if (!render->RegisterRenderBuffers(colorBuffers)) {
+      std::cerr << "RegisterRenderBuffers() returned false, cannot continue" << std::endl;
+      quit = true;
+    }
+
+    // Generate the vector of sphere locations we're going to use to control
+    // distortion correction.
+    std::vector<XY> spheres;
+    const size_t numSpheres = 1;
+    const double sphereSpace = 1.0 / (numSpheres);
```

Comment:
looks like you are just dividing here by 1 so sphereSpace still remains 1.0 (unless you will be changing numSpheres in future)